### PR TITLE
Add Support for XinFin ( XDC ) Network

### PR DESCRIPTION
**Pull request by AnilChinchawale**
Website: (https://www.xinfin.network / https://xinfin.org)

Block Explorer1 : (https://explorer.xinfin.network)

Block Explorer2 : (https://xdc.network)

Block Explorer3 : (https://xdc.blocksscan.io)

Slip 044 index : 550

ChainID / NetworkID : 50

### CODE REVIEW

**Comment by TamtamHero** [comment]
Can you please give us rights to push on this PR ?

File: Makefile [hunk 1]
```diff
@@ -31,7 +31,7 @@ APP_LOAD_PARAMS += --path "1517992542'/1101353413'"
 APPVERSION_M=1
 APPVERSION_N=9
 APPVERSION_P=14
-APPVERSION=$(APPVERSION_M).$(APPVERSION_N).$(APPVERSION_P)
+APPVERSION=$(APPVERSION_M).$(APPVERSION_N).$(APPVERSION_P)-xdc
 APP_LOAD_FLAGS= --appFlags 0x240 --dep Ethereum:$(APPVERSION)
```
Comment (jibeee) [inline Makefile:35]: Remove this before merging in `app-ethereum` repository.

**Review by jibeee — CHANGES_REQUESTED** [review]
Modified version label must be removed before merging this PR.

File: Makefile (hunk 1 above)
Comment (AnilChinchawale) [reply Makefile]: The modified version label has been removed, Please check now.

**Comment by Draghtnod** [comment]
@AnilChinchawale Looks like an upstream-merge is required. Shyft was faster.
Also, the other PRs suggest that you'd have to poke the bear by re-requesting a review to get your changes noticed.

**Comment by AnilChinchawale** [comment]
@Draghtnod Required changes has been updated, Can you check now.

**Review by jibeee — APPROVED** [review]

File: src/chainConfig.h [hunk 2]
```diff
@@ -59,6 +59,8 @@ typedef enum chain_kind_e {
     CHAIN_KIND_SONGBIRD,
     CHAIN_KIND_POLYGON,
     CHAIN_KIND_SHYFT
```
Comment (Draghtnod) [inline src/chainConfig.h:61]: There is a comma missing..

File: src/main.c [hunk 3]
```diff
@@ -269,6 +269,11 @@ tokenDefinition_t *getKnownToken(uint8_t *contractAddress) {
         case CHAIN_KIND_SHYFT:
             numTokens = NUM_TOKENS_SHYFT;
             break;
+        case CHAIN_KIND_XDCNETWORK:
+            numTokens = NUM_TOKENS_XDCNETWORK;
+            break;
+        case CHAIN_KIND_APOTHEMNETWORK:
+            numTokens = NUM_TOKENS_APOTHEMNETWORK;
```
Comment (Draghtnod) [inline src/main.c:276]: And a break here, just for consistency..

**Review by Draghtnod — COMMENTED** [review]
Found 2 minor issues.

File: src/chainConfig.h (hunk 2 above)
Comment (AnilChinchawale) [reply src/chainConfig.h]: Changes have been added.

File: src/main.c (hunk 3 above)
Comment (AnilChinchawale) [reply src/main.c:276]: Changes have been added.

File: src/main.c [hunk 4]
```diff
@@ -380,6 +386,12 @@ tokenDefinition_t *getKnownToken(uint8_t *contractAddress) {
             case CHAIN_KIND_SHYFT:
                 currentToken = (tokenDefinition_t *) PIC(&TOKENS_SHYFT[i]);
                 break;
+            case CHAIN_KIND_XDCNETWORK:
+                currentToken = (tokenDefinition_t *)PIC(&TOKENS_XDCNETWORK[i]);
+                break
```
Comment (Draghtnod) [inline src/main.c:391]: Missing semicolon here.

File: src/main.c [hunk 5]
```diff
@@ -380,6 +386,12 @@ tokenDefinition_t *getKnownToken(uint8_t *contractAddress) {
...
+            case CHAIN_KIND_APOTHEMNETWORK:
+                currentToken = (tokenDefinition_t *)PIC(&TOKENS_APOTHEMNETWORK[i]);
+                break
```
Comment (Draghtnod) [inline src/main.c:394]: And here. :)

**Review by Draghtnod — COMMENTED** [review]
Great! Just noticed another 2 missing semicolons. Sorry. :facepalm:

**Review by Draghtnod — APPROVED** [review]
Looks good to me. What do you say @jibeee ?

**Comment by Draghtnod** [comment]
I have ran a manual lint-job to get the log of the linter. It outputs a lot of unrelated stuff. The difference of the linter log between master and this PR is this:
```
< --- ./src/main.c      (original)
< +++ ./src/main.c      (reformatted)
< @@ -387,10 +387,10 @@
<                  currentToken = (tokenDefinition_t *) PIC(&TOKENS_SHYFT[i]);
<                  break;
<              case CHAIN_KIND_XDCNETWORK:
< -                currentToken = (tokenDefinition_t *)PIC(&TOKENS_XDCNETWORK[i]);
< +                currentToken = (tokenDefinition_t *) PIC(&TOKENS_XDCNETWORK[i]);
<                  break;
<              case CHAIN_KIND_APOTHEMNETWORK:
< -                currentToken = (tokenDefinition_t *)PIC(&TOKENS_APOTHEMNETWORK[i]);
< +                currentToken = (tokenDefinition_t *) PIC(&TOKENS_APOTHEMNETWORK[i]);
<                  break;
<          }
<          if (memcmp(currentToken->address, tmpContent.txContent.destination, ADDRESS_LENGTH) == 0) {
```
So there is a space missing before the `PIC(...`. Maybe this is why the Lint-Job fails.

**Comment by Draghtnod** [comment]
:tada: It worked! Now it is just out of date again.

**Comment by AnilChinchawale** [comment]
@Draghtnod I have updated the branch, Can you review it.

**Review by Draghtnod — APPROVED** [review]
- No issues found in code :heavy_check_mark:
- Linter did not complain :heavy_check_mark:
- Build for Nano S on default chain successful :heavy_check_mark:
- Build for Nano S on xdcnetwork chain successful :heavy_check_mark:
- Build for Nano S on apothemnetwork chain successful :heavy_check_mark:
- Build for Nano X on default chain successful :heavy_check_mark:
- Build for Nano X on xdcnetwork chain successful :heavy_check_mark:
- Build for Nano X on apothemnetwork chain successful :heavy_check_mark:

Are we there yet @jibeee ? :smile:

**Pull request closed without merging**